src/vs/vscode.d.ts

```diff
@@ -1812,6 +1812,56 @@ declare module 'vscode' {
 	 */
 	export type ProviderResult<T> = T | undefined | null | Thenable<T | undefined | null>;
 
+	/**
+	 * Scope of a code action.
+	 *
+	 * Specifies the type of a code action. A scope is a hierarchical list of identifiers separated by `.`, e.g. `"refactoring.extract.function"`.
+	 */
+	export class CodeActionScope {
+		/**
+		 * Empty scope.
+		 */
+		public static readonly Empty: CodeActionScope;
+
+		/**
+		 * Base scope for refactoring actions.
+		 */
+		public static readonly Refactor: CodeActionScope;
+
+		/**
+		 * Base scope for refactoring extraction actions.
+		 */
+		public static readonly RefactorExtract: CodeActionScope;
+
+		/**
+		 * Base scope for refactoring extraction inline.
+		 */
+		public static readonly RefactorInline: CodeActionScope;
+
```

Comment:
We should also have `static readonly Rewrite` for things like 'Convert if-else into ternary statement' etc, etc

Reply:
and also maybe `CleanUp` category?